light/beacon/api/rest_api.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more detaiapi.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	cbeacon "github.com/ethereum/go-ethereum/core/beacon"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/light/beacon"
+)
+
+// RestApi implements LightClientApi by requesting information from a beacon node REST API.
+// Note: all required API endpoints are currently only implemented by Lodestar.
+type RestApi struct {
+	Url string
+}
```

Comment:
IMO the name `RestApi` does not properly describe what this thing is representing - I'm thinking that what this is representing, is a Remote CL-node. 
So I think you should rename it to `RemoteCL`, or `RemoteBeacon`.

Reply:
I also think you might extend it a bit, like I did here: https://github.com/holiman/factor/blob/main/lib/remote_cl.go#L30 

That means it can be used with custom headers. If we want to try these things with the nodes that the devops-team runs, we need to use special headers in order to gain access to the couldflare environment.

Reply:
I named it `BeaconLightApi` because it's not always remote (in the blsync use case it typically is, but the same code will be used for LES servers that can and should probably use local beacon nodes).
I'll also do the custom header thing.

Reply:
custon HTTP header is also done